File: java/src/com/db4o/dg2db4o/chapter8/RunEmbeddedListClient.java

package com.db4o.dg2db4o.chapter8;

import com.db4o.*;

public class RunEmbeddedListClient implements Runnable {
    ObjectServer server;
    ObjectContainer rClient = null;
    private boolean stop = false;
    
    public RunEmbeddedListClient(ServerRegistry sr) {
        server = sr.getServer("myserver");
    }
    
    public void run() {
        synchronized(this){
            try {
                rClient = server.openClient();
                while (!stop) { 
                    listResult(rClient.get(new Person()));
                    Thread.sleep(15000);
                }
            } catch (Exception e) {
                e.printStackTrace();
                System.out.println("\nLISTCLIENT: server not available");
            } finally {
                rClient.close();
            }
        }
    }
    
    public static void listResult(ObjectSet result) {
        System.out.println("\nLISTCLIENT: Listing...");
        while (result.hasNext()) {
            System.out.println("LISTCLIENT:" + result.next());
        }
    }
}

File: csharp/Chapter08/RunEmbeddedListClient.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading;
using com.db4o;

namespace com.db4o.dg2db4o.chapter8
{
    class RunEmbeddedListClient
    {
        private ObjectServer server;
        private ObjectContainer lClient = null;
        private bool stop = false;

        public RunEmbeddedListClient(ServerRegistry sr)
        {
            server = sr.GetServer("myserver");
        }

        public void Run()
        {
            lock (this)
            {
                Console.WriteLine("Starting list client...");
                ServerRegistry sr2 = ServerRegistry.GetInstance();
                server = sr2.GetServer("myserver");
                try
                {
                    lClient = server.OpenClient();
                    while (!stop)
                    { // Your stop condition here...
                        ListResult(lClient.Get(new Person()));
                        Monitor.Wait(this, 10000);
                    }
                }
                catch (ThreadInterruptedException tie)
                {
                    Console.WriteLine("Thread Error!" + tie);
                }
                catch (Exception e)
                {
                    Console.WriteLine("Server Error!" + e);
                }
                finally
                {
                    lClient.Close();
                }
            }
        }

        private static void ListResult(ObjectSet result)
        {
            Console.WriteLine("\nLISTCLIENT: Listing...");
            while (result.HasNext())
            {
                Console.WriteLine("LISTCLIENT:" + result.Next());
            }
        }

    }
}
